Examples/Algorithms/Digitization/src/SmearingAlgorithm.cpp

```diff
@@ -0,0 +1,95 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Digitization/SmearingAlgorithm.hpp"
+#include "ActsExamples/EventData/GeometryContainers.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+#include <Acts/Geometry/GeometryIdentifier.hpp>
+#include <Acts/Geometry/TrackingGeometry.hpp>
+
+#include <stdexcept>
+
+ActsExamples::SmearingAlgorithm::SmearingAlgorithm(
+    ActsExamples::SmearingAlgorithm::Config cfg, Acts::Logging::Level lvl)
+    : ActsExamples::BareAlgorithm("SmearingAlgorithm", lvl),
+      m_cfg(std::move(cfg)) {
+  if (m_cfg.inputSimulatedHits.empty()) {
+    throw std::invalid_argument("Missing input hits collection");
+  }
+  if (m_cfg.outputMeasurements.empty()) {
+    throw std::invalid_argument("Missing output measurement collection");
+  }
+  if (!m_cfg.randomNumbers) {
+    throw std::invalid_argument("Missing random numbers tool");
+  }
+
+  // fill the digitizables map to allow lookup by geometry id only
+  m_cfg.trackingGeometry->visitSurfaces([this](const Acts::Surface* surface) {
+    this->m_dSurfaces.insert_or_assign(surface->geometryId(), surface);
+  });
+}
+
+ActsExamples::ProcessCode ActsExamples::SmearingAlgorithm::execute(
+    const AlgorithmContext& ctx) const {
+  if (not m_cfg.configured) {
+    ACTS_FATAL("Smearing Algorithm is misconfigured. Aborting.");
+    return ProcessCode::ABORT;
+  }
+
+  const auto& hits =
+      ctx.eventStore.get<SimHitContainer>(m_cfg.inputSimulatedHits);
+
+  ActsExamples::GeometryIdMultimap<Acts::FittableMeasurement<DigitizedHit>>
+      measurements;
```

Comment:
Consider reserving enough space to fit all hits so you do not need to reallocate.

Reply:
done.